---
title: "Vignette"
author: "Tate Andrew Keller"
date: "4/9/2020"
output: html_document
---

# Plot Feature Counts

```{r}

library(magrittr)
library(data.table)
library(ggplot2)
library(reshape2)

genes <- data.table::fread(file = '../featureCounts/featureCounts.txt.summary', header = TRUE)%>% as.data.frame
genes
originalNames <- names(genes)
names(genes) <- gsub(".*(Acute_Fatal|Acute_Survivor)(_[0-9]+).*", "\\1\\2", originalNames)
newNames = names(genes)

assigned = genes[genes$Status == 'Assigned',]
unassigned_nofeatures = genes[genes$Status == 'Unassigned_NoFeatures',]
unassigned_ambiguity = genes[genes$Status == 'Unassigned_Ambiguity',]

df = rbind(assigned,unassigned_nofeatures,unassigned_ambiguity)

df = transpose(df)
names(df) <- df[1,]
df <- df[-1,]

df$types = newNames[-1]

dfm <- melt(df[,c('types','Assigned','Unassigned_NoFeatures','Unassigned_Ambiguity')],id.vars = 1)

ggplot(data = dfm, aes(x = types, y = as.integer(value), fill = variable)) + geom_bar(stat='identity', position='dodge')



```

# Full Feature Counts

```{r}
genesFull <- data.table::fread(file = '../featureCounts/featureCounts.txt', header = TRUE)%>% as.data.frame
originalNames <- names(genesFull)
names(genesFull) <- gsub(".*(Acute_Fatal|Acute_Survivor)(_[0-9]+).*", "\\1\\2", originalNames)
newNames = names(genesFull)
str(genesFull)
```

# DESeq object

```{r}
library(DESeq2)
row.names(genesFull) <- make.names(genesFull$Geneid)
genesFull <- genesFull[ , -c(1:6)]
sample_info <- DataFrame(condition = gsub("_[0-9]+", "", names(genesFull)),row.names = names(genesFull) )
DESeq.ds <- DESeqDataSetFromMatrix(countData = genesFull,colData = sample_info,design = ~ condition)
DESeq.ds
head(counts(DESeq.ds))
```

# Remove genes with no reads

```{r}
keep_genes <- rowSums(counts(DESeq.ds)) > 0
DESeq.ds <- DESeq.ds[ keep_genes, ]
```


```{r}
DESeq.ds <- estimateSizeFactors(DESeq.ds) 
plot( sizeFactors(DESeq.ds), colSums(counts(DESeq.ds)),ylab = "library sizes", xlab = "size factors", cex = .6 )
par(mfrow=c(1,2))
counts.sf_normalized <- counts(DESeq.ds, normalized=TRUE)
boxplot(counts.sf_normalized, main = "SF normalized", cex = .6)
boxplot(counts(DESeq.ds), main = "read counts only", cex = .6)
```

```{r}
par(mfrow=c(1,2))
boxplot(log2(counts(DESeq.ds)+1), notch=TRUE, main = "Non-normalized read counts", ylab="log2(read counts)", cex = .6)
boxplot(log2(counts(DESeq.ds, normalize= TRUE) +1), notch=TRUE, main = "Size-factor-normalized read counts", ylab="log2(read counts)", cex = .6)
```

```{r}
log.counts <- log2(counts(DESeq.ds, normalized = FALSE) + 1)
assay(DESeq.ds, "log.counts") <- log2(counts(DESeq.ds, normalized = FALSE) + 1)
log.norm.counts <- log2(counts(DESeq.ds, normalized=TRUE) + 1)
assay(DESeq.ds, "log.norm.counts") <- log.norm.counts
log.norm.counts <- log2(counts(DESeq.ds, normalized=TRUE) + 1)
msd_plot <- vsn::meanSdPlot(log.norm.counts,ranks=FALSE,plot = FALSE)
msd_plot$gg + ggtitle("Sequencing depth normalized log2(read counts)") + ylab("standard deviation")
```


```{r}
DESeq.vst <- vst(DESeq.ds, blind = TRUE)
par(mfrow=c(1,2))
plot(log.norm.counts[,1:2], cex=.1,main = "size factor and log2-transformed")
plot(assay(DESeq.vst)[,1],assay(DESeq.vst)[,2],cex=.1, main = "vst transformed", xlab = colnames(assay(DESeq.vst[,1])), ylab = colnames(assay(DESeq.vst[,2])) )
```

```{r}
vst.norm.counts <- assay(DESeq.vst)
msd_plot <- vsn::meanSdPlot( vst.norm.counts, ranks=FALSE, plot = FALSE)
msd_plot$gg + ggtitle("vst transformation") + coord_cartesian(ylim = c(0,3))
```

# PCA

```{r fig.width=20, fig.height=30}
vst.norm.counts = vst.norm.counts[ , order(colnames(vst.norm.counts))]
corr_coeff <- cor(vst.norm.counts, method = "pearson")
as.dist(1-corr_coeff, upper = TRUE) %>% as.matrix %>% pheatmap::pheatmap(., main = "Pearson correlation")
```

```{r}
#par(mfrow=c(1,2))
as.dist(1 - corr_coeff) %>% hclust %>% plot( ., labels = colnames(vst.norm.counts), main = "vst transformed read counts")
as.dist( 1 - cor(log.norm.counts, method = "pearson")) %>% hclust %>% plot( ., labels = colnames(log.norm.counts), main = "no vst")
```


```{r}
rv <- rowVars(assay(DESeq.vst))
top_variable <- order(rv, decreasing = TRUE)[seq_len(500)]
pca <- prcomp(t(assay(DESeq.vst)[top_variable, ]))
head(pca$x)
plotPCA(DESeq.vst)
```

# DE Analysis

```{r}
DESeq.ds$condition <- relevel(DESeq.ds$condition, ref="Acute_Survivor")
DESeq.ds <- DESeq(DESeq.ds)
DESeq.ds <- estimateSizeFactors(DESeq.ds)
DESeq.ds <- estimateDispersions(DESeq.ds)
DESeq.ds <- nbinomWaldTest(DESeq.ds)
```

```{r}
DGE.results <- results(DESeq.ds, independentFiltering = TRUE, alpha = 0.05)
```

```{r}
hist(DGE.results$padj, col="grey", border="white", xlab="", ylab="", main="frequencies of padj\n(all genes)", cex = 0.4)
DGE.results.sorted <- DGE.results[order(DGE.results$padj),]
```

```{r}
table(DGE.results$padj < 0.05)
head(DGE.results.sorted)
plotCounts(DESeq.ds, gene="ORM1", normalized = TRUE)
plotCounts(DESeq.ds, gene="C1S", normalized = TRUE)
plotCounts(DESeq.ds, gene="TPMT", normalized = TRUE)
plotCounts(DESeq.ds, gene="FGFBP2", normalized = TRUE)
plotCounts(DESeq.ds, gene="DST", normalized = TRUE)
plotCounts(DESeq.ds, gene="AEBP1", normalized = TRUE)
plotCounts(DESeq.ds, gene="HSPA4L", normalized = TRUE)

```

```{r}
DGEgenes <- rownames(subset(subset(DGE.results.sorted, padj < 0.05), abs(log2FoldChange) > 2.5 ))
vst.dge <- DESeq.vst[DGEgenes,] %>% assay
library(pheatmap)
```

```{r fig.width=6, fig.height=6}
pheatmap(vst.dge, scale="row", show_rownames = TRUE, main = "DGE (row-based z-score)")
```

```{r}
#DESeq.ds
#DGE.results
DGE.results.sortedbyLog2 <- DGE.results[order(DGE.results$log2FoldChange),]
genesList = subset(subset(DGE.results.sortedbyLog2, padj < 0.05), abs(log2FoldChange) > 1.6 )
genesList
```

```{r}
genesList
#plotCounts(DESeq.ds, gene=which.max(genesList$log2FoldChange),  normalized = TRUE)
#plotCounts(DESeq.ds, gene=which.min(genesList$log2FoldChange),  normalized = TRUE)

# More Survival
plotCounts(DESeq.ds, gene="PRSS8", normalized = TRUE) # Sodium channel regulation https://www.nature.com/articles/hr20096
plotCounts(DESeq.ds, gene="CKB", normalized = TRUE) # Brain cell homeostasis? 
plotCounts(DESeq.ds, gene="UCA1", normalized = TRUE) # upregulated in bladder cancer? https://clincancerres.aacrjournals.org/content/12/16/4851

#More fatal
plotCounts(DESeq.ds, gene="ORM1", normalized = TRUE) # immunosuppression  https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=ShowDetailView&TermToSearch=5004
plotCounts(DESeq.ds, gene="IL6", normalized = TRUE) # ctyokine storm? 
plotCounts(DESeq.ds, gene="C7", normalized = TRUE) # Involved in lysis of pathogens, interesting that it is more present in fatal cases # https://www.genecards.org/cgi-bin/carddisp.pl?gene=C7
plotCounts(DESeq.ds, gene="SAA1", normalized = TRUE) # https://www.nejm.org/doi/10.1056/NEJM199902113400607
plotCounts(DESeq.ds, gene="CCN1", normalized = TRUE) # big range of functions, including immflamation, could be interesting to look into further https://www.genecards.org/cgi-bin/carddisp.pl?gene=CCN1
```


```{r}
plotMA(DGE.results, alpha = 0.05, main = "Test: p.adj.value < 0.05", ylim = c(-4,4))
```

```{r}
library(EnhancedVolcano)
vp1 <- EnhancedVolcano(DGE.results, lab = rownames(DGE.results), x = 'log2FoldChange', y = 'padj', pCutoff = 0.05, title = "Fatal / Survivor", ylim = c(1,7), xlim = c(-5,5))
print(vp1)
```

```{r}
plotCounts(DESeq.ds, gene="PRSS8", normalized = TRUE,transform = TRUE)
plotCounts(DESeq.ds, gene="CCN1", normalized = TRUE,transform = TRUE)
plotCounts(DESeq.ds, gene="CKB", normalized = TRUE,transform = TRUE)
plotCounts(DESeq.ds, gene="SAA1", normalized = TRUE,transform = TRUE)
plotCounts(DESeq.ds, gene="ORM1", normalized = TRUE,transform = TRUE)

```

```{r}
library(DESeq2)
DGE.results.shrnk <- lfcShrink(DESeq.ds, coef = 2, type = "apeglm")
gene.vector <- row.names(DGE.results.shrnk) %in% DGEgenes %>% as.integer
names(gene.vector) <- row.names(DGE.results.shrnk)
```

```{r}
library(clusterProfiler)
library(org.Hs.eg.db)

DGEgenes <- rownames(subset(DGE.results.sorted, padj < 0.05))

anno.DGE <- select(org.Hs.eg.db,
keys = DGEgenes, # original rownames of DGE results
keytype="SYMBOL", # our rownames are ORF identifiers
columns=c("ENSEMBL","GENENAME")) 

anno.DGE

ego = enrichGO(gene = DGEgenes,
               OrgDb         = org.Hs.eg.db,
               keyType = 'SYMBOL',
               ont = "BP",
               pAdjustMethod = "BH",
               readable      = TRUE)

ego

```



```